docs/functionalabstractions/concretetypes/equal.md

```diff
@@ -3,4 +3,135 @@ id: equal
+
+Using `==` we can compare any two values for equality, even if they are unrelated types. For example, this code has a bug.
+
+```scala mdoc:warn
+final case class SequenceNumber(value: Int)
+
+def isInitial(sequenceNumber: SequenceNumber): Boolean =
+  sequenceNumber == 0
+```
+
+We're trying to determine if this is the initial sequence number , but we're comparing the `sequenceNumber`, which is a `SequenceNumber`, with `0`, which is an `Int`. So this comparison will always be false even if the `sequenceNumber` is `SequenceNumber(0)` which is what we are trying to test for.
+
+This snippet may generate a warning but it is still valid Scala code.
+
+In contrast, ZIO Prelude's `Equal` abstraction catches the bug and prevents this code from even compiling.
+
+```scala mdoc:reset:fail
+import zio.prelude._
+
+final case class SequenceNumber(value: Int)
+
+object SequenceNumber {
+  implicit val SequenceNumberEqual: Equal[SequenceNumber] =
+    Equal.default
+}
+
+def isInitial(sequenceNumber: SequenceNumber): Boolean =
+  sequenceNumber === 0 // does not compile
+```
+
+To get the code to compile we have to fix the bug.
+
+```scala mdoc
+import zio.prelude._
+
+case class SequenceNumber(value: Int)
+
+object SequenceNumber {
+
+  val initial: SequenceNumber =
+    SequenceNumber(0)
+
+  implicit val SequenceNumberEqual: Equal[SequenceNumber] =
+    Equal.default
+}
+
+def isFirst(sequenceNumber: SequenceNumber): Boolean =
+  sequenceNumber === SequenceNumber.initial // okay
+```
+
+A second advantage is catching bugs when we try to compare two values of a type that doesn't have a meaningful notion of equality. For example, it doesn't make sense to compare two Scala functions for equality, but we can still do it with `==`.
+
+In contrast, comparing two functions with `===` will result in a compilation error, preventing us from making this mistake.
+
+A third advantage of the `Equal` abstraction is that it lets us use our own notion of equality for data types that are not under our control. We may be working with data types from other libraries that do not implement a reasonable notion of equality, for example using reference equality instead of value equality.
+
+If we are using `==` we can't do anything about that because `==` is always based on how the data type implements `equals`. But if we are using `===` from ZIO Prelude we can define our own `Equal` instance that implements a more sensible notion of equality.
+
+## Defining Equal Instances
+
+Equal instances are automatically available for data types from the Scala library and ZIO with a meaningful notion of equality, as well as data types made up of those types.
+
+We can always check if an instance of a type class exists for a given data type by using the `apply` operator to tell the Scala compiler to find the appropriate instance for us. This will result in a compilation error if the instance cannot be found.
+
+```scala mdoc
+Equal[List[Either[String, Int]]] // okay
```

Comment:
add a failing example for instance a case class wrapping a function? Since functions have no `Equal` a case class wrapping a function should not.

Reply:
Added.